build/notebooks/table_variants_missense.md

```diff
@@ -5070,7 +5070,246 @@ tbl_variants_selected.displayall(tr_style=tr_style, td_styles=td_styles)
+# \texttt{L995F} & 
+# \texttt{L995S} \\
+# \midrule
+# """
+# template = r"""
+# {substitution} & {AGAP004707-RA} & NA & {AF_AOM} & {AF_BFM} & {AF_GNS} & {AF_BFS} & {AF_CMS} & {AF_GAS} & {AF_UGS} & {AF_KES} & {AF_GWA} & NA & NA \\
+# """
+# epilogue = r"""
+# \bottomrule
+# \end{tabular}
+# """
+# tbl_variants_display.totext('../tables/variants_missense.tex', 
+#                             encoding='ascii',
+#                             prologue=prologue, 
+#                             template=template,
+#                             epilogue=epilogue)
+```
+
+
+```python
+# !cat ../tables/variants_missense.tex
+```
+
+# add housefly numbering
+
+
+```python
+#to easily split the codon numbers from the codon letters
+import re
+```
+
+
+```python
+#load the codon map from the blog post (with the header info removed)
+md_tbl = etl.fromtsv('/home/chris/Git/agam-vgsc-report/data/domestica_gambiae_map.txt')
+
+#get codons
+dom = list(md_tbl['domestica_codon'])
+ano = list(md_tbl['gambiae_codon'])
+```
+
+
+```python
+#make a dictionary with a as the key and d as the value
+map_dict = {a: d for a, d in zip(ano, dom)}
+```
+
+
+```python
+#get the snpeff annotations
+gam_cod = list(tbl_variants_selected['SNPEFF_HGVS_p'])
+```
+
+
+```python
+#ditch the codon letters - using re the regex module - LOVELY
+gam_cod_cl = []
+r = re.compile("([a-zA-Z]+)([0-9]+)")
+for c in gam_cod:
+    if c:
+        d = c[2:10]
+        m = r.match(d)
+        g = m.group(2)
+        gam_cod_cl.append(g)
+len(gam_cod_cl)
+```
+
+
+
+
+    20
+
+
+
+
+```python
+MD = [map_dict[c] for c in gam_cod_cl]
+MD
+```
+
+
 
+
+    ['261',
+     '410',
+     '.',
+     '508',
+     '810',
+     '1014',
+     '1014',
+     '1133',
+     '1262',
+     '1532',
+     '1575',
+     '1602',
+     '1608',
+     '1751',
+     '1858',
+     '1873',
+     '1879',
+     '1879',
+     '1939',
+     '1945']
+
+
+
+
+```python
+#get the musca codon letter at these positions and add
+fs = pyfasta.Fasta('/home/chris/Git/agam-vgsc-report/data/domestica_gambiae_PROT_MEGA.fas')
+```
+
+
+```python
+#grab the right sample
+dom = fs.get('domestica_vgsc')
+```
+
+
+```python
+#remove the '-' from the aligned fasta so the numbering makes sense
+dom_fix = [p for p in dom if p != '-']
+#check
+dom_fix[261-1],dom_fix[1945-1]
+```
+
+
+
+
+    ('R', 'I')
+
+
+
+
+```python
+#add them to the position
+```
+
+
+```python
+MD_fix = []
+for p in MD:
+    if p == '.':
+        MD_fix.append(p)
+    if p != '.':
+        MD_fix.append(dom_fix[int(p)-1]+p)
+```
+
+
+```python
+MD_fix
+```
+
+
+
+
+    ['R261',
+     'V410',
+     '.',
+     'M508',
+     'T810',
+     'L1014',
+     'L1014',
+     'K1133',
+     'I1262',
+     'I1532',
+     'N1575',
+     'E1602',
+     'K1608',
+     'A1751',
+     'V1858',
+     'I1873',
+     'P1879',
+     'P1879',
+     'A1939',
+     'I1945']
+
+
+
+
+```python
+#because of the multi-allelics there are a couple of repeated jobbys, do these by hand...
+#also add the 
```

Comment:
See comment above, should be possible to get this without having to fix by hand.